### PR TITLE
Modernize build

### PR DESCRIPTION
**Pull request by ermshiperete**
- Add GitHub actions
- Use NUnit 3
- Update build dependencies
- Move Directory.Build.props to source root
- Allow to build with dotnet

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/l10nsharp/102)

### CODE REVIEW

File: src/L10NSharpTests/LocalizationExtenderTests.cs [hunk 1]
```diff
@@ -185,7 +190,8 @@ public void PrepareDataGridViewColumnsTests()
 		/// group for each.
 		/// </summary>
 		/// ------------------------------------------------------------------------------------
-		[Test, Ignore("The test body was commented out before I got here, why run it?")]
+		[Test]
+		[Ignore("The test body was commented out before I got here, why run it?")]
```
Comment (papeh) [inline src/L10NSharpTests/LocalizationExtenderTests.cs:194]: If it's been commented out that long, maybe it should be deleted.

**Review by papeh — APPROVED** [review]

File: src/CheckOrFixXliff/Program.cs [hunk 2]
```diff
@@ -130,7 +131,7 @@ private static void FixBrokenTransUnit(XElement tu, XmlNamespaceManager namespac
 			var target = tu.XPathSelectElement("x:target", namespaceManager);
 			if (source.HasElements || target.HasElements)
 			{
-				Console.WriteLine("Cannot fix {0} because the translated material contains XML elements", tuid);
+				Console.WriteLine(@"Cannot fix {0} because the translated material contains XML elements", tuid);
```
Comment (papeh) [inline src/CheckOrFixXliff/Program.cs:134]: I don't think this `@` is changing anything (here and throughout).

File: src/ExtractXliff/Program.cs [hunk 3]
```diff
@@ -91,26 +92,28 @@ static void Main(string[] args)
...
 						if (_verbose)
-							Console.WriteLine($"Method {methodNameSpec.Item2}.{methodNameSpec.Item3} in {asm.GetName().FullName} will be treated as a localization method.");
+							Console.WriteLine($@"Method {methodNameSpec.Item2}.{methodNameSpec.Item3} in {asm.GetName().FullName} will be treated as a localization method.");
 						_additionalLocalizationMethodNames.RemoveAt(index--);
 					}
 					catch (Exception e)
 					{
 						if (_verbose)
-							Console.WriteLine("Error using reflection on {asm.GetName().FullName} to get type {methodNameSpec.Item2} or method {methodNameSpec.Item3}:" + e.Message);
+							Console.WriteLine($@"Error using reflection on {asm.GetName().FullName} to get type {methodNameSpec.Item2} or method {methodNameSpec.Item3}:" + e.Message);
```
Comment (papeh) [inline src/ExtractXliff/Program.cs:101]: e.Message could be included in the interpolated string

File: src/CheckOrFixXliff/Program.cs (hunk 2 above)
Comment (ermshiperete) [reply src/CheckOrFixXliff/Program.cs:134]: It marks it as verbatim string and so R# knows that it's not a localizable string.

File: src/ExtractXliff/Program.cs (hunk 3 above)
Comment (ermshiperete) [inline src/ExtractXliff/Program.cs:101]:
```suggestion
							Console.WriteLine($@"Error using reflection on {asm.GetName().FullName} to get type {methodNameSpec.Item2} or method {methodNameSpec.Item3}: {e.Message}");
```

File: src/CheckOrFixXliff/Program.cs (hunk 2 above)
Comment (papeh) [reply src/CheckOrFixXliff/Program.cs:134]: Thanks. That's what I thought, but the option to ignore verbatim strings had been turned off in FieldWorks, so I had trained myself to stop doing that.

**Review by papeh — APPROVED** [review]

**Pull request merged**